dask/dataframe/core.py

```diff
@@ -390,13 +390,38 @@ def _loc(self, ind):
         if isinstance(ind, Series):
             return self._loc_series(ind)
         if self.known_divisions:
+            ind = self._maybe_time_slice(ind, kind='loc')
             if isinstance(ind, slice):
                 return self._loc_slice(ind)
             else:
                 return self._loc_element(ind)
         else:
             return map_partitions(try_loc, self, self, ind)
 
+    def _maybe_time_slice(self, ind, kind):
+        index = self._pd.index
+
+        if not isinstance(index, (pd.DatetimeIndex, pd.PeriodIndex)):
+            return ind
+
+        if isinstance(ind, slice):
+            if isinstance(ind.start, pd.compat.string_types):
+                start = index._maybe_cast_slice_bound(ind.start, 'left', kind)
+            else:
+                start = ind.start
+
+            if isinstance(ind.stop, pd.compat.string_types):
+                stop = index._maybe_cast_slice_bound(ind.stop, 'right', kind)
+            else:
+                stop = ind.stop
+            return slice(start, stop)
+
+        elif isinstance(ind, pd.compat.string_types):
+            start = index._maybe_cast_slice_bound(ind, 'left', 'loc')
+            stop = index._maybe_cast_slice_bound(ind, 'right', 'loc')
+            return slice(start, stop)
+        return ind
```

Comment:
OK. What do you think split index-related functions to separate file, ``indexing.py`` (same name as pandas)?